pkg/skaffold/render/renderer/render_mux.go

```diff
@@ -80,12 +80,25 @@ func (r RenderMux) Render(ctx context.Context, out io.Writer, artifacts []graph.
 		endTrace()
 	}
 	w, ctx = output.WithEventContext(ctx, out, constants.Render, constants.SubtaskIDNone)
-	for i := range r.gr.HookRunners {
-		if err := r.gr.HookRunners[i].RunPostHooks(ctx, w); err != nil {
-			return manifest.ManifestListByConfig{}, err
+
+	if len(r.gr.HookRunners) == 0 {
+		return allManifests, nil
+	}
+
+	updated := manifest.NewManifestListByConfig()
+	for _, name := range allManifests.ConfigNames() {
```

Comment:
hmmm, the previous behavior sounds more like a bug rather than a feature. In my opinion, the hook should be scoped, the hook should be running when the corresponding event triggered and should be able to access event data and only its own event data and optionally produces other data for next phases in the same pipeline to use.  Renders cannot access rendered results from other pipelines, it's more reasonable to let the hook behave the same,  I think.